src/test/java/ch/uzh/ifi/hase/soprafs21/controller/UsersApiControllerTest.java

```diff
@@ -0,0 +1,89 @@
+package ch.uzh.ifi.hase.soprafs21.controller;
+
+import ch.uzh.ifi.hase.soprafs21.rest.dto.UserLoginDto;
+import ch.uzh.ifi.hase.soprafs21.rest.dto.UserLoginPostDto;
+import ch.uzh.ifi.hase.soprafs21.service.JwtTokenUtil;
+import ch.uzh.ifi.hase.soprafs21.service.UserService;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.json.webtoken.JsonWebSignature;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+
+
+class UsersApiControllerTest {
```

Comment:
how do we test the application manually? due to the complexity, is there any way to disable the security, while implementing the other features, or at least to easily fake identities/tokens?

Reply:
For testing locally, we can add a particular path in below method of SecurityConfig.java, so it will ignore security for it
![image](https://user-images.githubusercontent.com/29889416/115106951-27518080-9f68-11eb-93a7-bdbdd54b8445.png)